### PR TITLE
allow editing in json editor for checked params

### PR DESCRIPTION
**Pull request by ThatGalNatalie**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
The JSON editor wasn't reflecting the checked/unchecked parameters. When modifying an existing schedule, the checked parameters should now be the only parameters that is displayed

### CODE REVIEW

**Comment by ThatGalNatalie** [comment]
@dylanbhughes

**Comment by dylanbhughes** [comment]
Having trouble looking in Netlify but the code makes sense to me!

**Comment by zhen0** [comment]
This addresses the issue it was asked to nicely but I fear that the JSON edit experience could still lead to mistakes.  If I go to the schedule parameters tab and immediately switch into JSON, add my parameters and then switch back to set the schedule, I'd miss that I need to check the box to include my parameters.  Could we add a JSON edit specific warning?

**Review by zhen0 — CHANGES_REQUESTED** [review]
I think this fixes the issue @ThatGalNatalie - thanks.  However, I think we risk wiping all a user's JSON input.  Can we add a warning that they also need to check the boxes?  Or remove the need to check boxes?

**Comment by znicholasbrown** [comment]
Previews enabled for this PR 👍

File: src/components/CustomInputs/DictInput.vue [hunk 1]
```diff
@@ -92,11 +95,13 @@ export default {
         this.$nextTick(() => {
           this.$refs['json-input'].validateJson()
         })
+      } else {
+        this.includedKeys = Object.keys(this.value)
       }
     },
-    includedKeys() {
+    includedKeys(val) {
+      this.jsonInput = val.length > 0 ? JSON.stringify(this.value) : '{}'
```
Comment (dylanbhughes) [inline src/components/CustomInputs/DictInput.vue:103]: Makes sense!

**Comment by ThatGalNatalie** [comment]
update:
Latest commit should allow you to edit in the JSON editor and the parameters should be checked/unchecked based on that. 

However, if you check a parameter then it will be added to the jsoninput but if you decide to uncheck a parameter it will not be reflected in the json editor. I think it has something to do with the `this.value` computed prop and how we’re checking what gets returned but can’t seem to wrap my head around it and need some help on that. 

Tested the run page briefly and it seems to not be affected by this change but will need to do a more thorough test

File: src/components/CustomInputs/DictInput.vue [hunk 2]
```diff
@@ -92,11 +95,13 @@ export default {
         this.$nextTick(() => {
           this.$refs['json-input'].validateJson()
         })
+      } else {
+        this.includedKeys = Object.keys(this.value)
```
Comment (znicholasbrown) [inline src/components/CustomInputs/DictInput.vue:99]:
```suggestion
        try {
          this.includedKeys = Object.keys(JSON.parse(this.jsonInput))
        } catch {
          this.includedKeys = Object.keys(this.value)
        }
```

File: src/components/CustomInputs/DictInput.vue [hunk 3]
```diff
@@ -70,7 +70,10 @@ export default {
     value() {
       const dict = {}
       this.keys.forEach((k, i) => {
-        if (k && (!this.includeCheckbox || this.includedKeys.includes(k))) {
+        if (
+          (k && (!this.includeCheckbox || this.includedKeys.includes(k))) ||
+          Object.keys(JSON.parse(this.jsonInput)).includes(k)
+        ) {
           dict[k] = this.values[i]
         }
```
Comment (znicholasbrown) [inline src/components/CustomInputs/DictInput.vue:78]:
```suggestion
        if (
          (k && (!this.includeCheckbox || this.includedKeys.includes(k))) ||
          (this.json && Object.keys(JSON.parse(this.jsonInput)).includes(k))
        ) {
          dict[k] = this.values[i]
        }
```

**Review by znicholasbrown — CHANGES_REQUESTED** [review]
2 suggestions which I think should fix the issue you're seeing

File: src/components/CustomInputs/DictInput.vue (hunk 3 above)
Comment (ThatGalNatalie) [reply src/components/CustomInputs/DictInput.vue:78]: ahhhh it was checking if they were on the json editor then push the keys/values but if not push the keys/values from the checkboxes

**Review by zhen0 — COMMENTED** [review]
One small suggestion

File: src/components/CustomInputs/DictInput.vue [hunk 4]
```diff
@@ -108,10 +113,23 @@ export default {
...
-        )
+        if (this.includeCheckbox) {
+          Object.keys(json).forEach(k => {
+            let i = this.keys.indexOf(k)
+
+            if (i > -1) {
+              this.values[i] = json[k]
+            } else {
+              this.keys.push(k)
+              this.values.push(json[k])
```
[the remark this answers is not in the document]
Comment (ThatGalNatalie) [reply src/components/CustomInputs/DictInput.vue]: just pushed a new commit for this

**Review by znicholasbrown — APPROVED** [review]
This LGTM @ThatGalNatalie 👍

**Pull request merged**